src/views/CategoryCreator/CategoryCreator.jsx

```diff
@@ -0,0 +1,127 @@
+import React from 'react'
+import styled from 'styled-components'
+import { useFormik } from 'formik';
+import { useQuery } from 'react-query';
+import Input from '@material-ui/core/Input';
+import Button from '@material-ui/core/Button';
+import Select from '@material-ui/core/Select';
+import { Alert } from '@material-ui/lab';
+import paths from 'constants/api';
+import CenteredContainer from 'components/CenteredContainer'
+
+const StyledForm = styled.form`
+    display:flex;
+    flex-direction: column;
+    justify-content: center;
+
+    @media (min-width: 1024px){
+        width: 50%;
+    }
+`;
+
+const CategoryCreator = () => {
+    const getBlogs = async () => {
+        const url = `${paths.BLOGS}`;
+
+        const headers = {
+            'Content-Type': 'application/json',
+        };
+
+        const response = await fetch(url, { headers, method: 'GET' });
+
+        if (response.status !== 200) {
+            throw new Error('błąd');
+        }
+
+        return response.json();
+    }
+
+    const { data, isError } = useQuery('blogs', getBlogs)
+
+    const formik = useFormik({
+        initialValues: {
+            title: '',
+            blog: 0,
+            description: '',
+            cover_image: '',
+            header_image: ''
+        },
+        onSubmit: () => {
+            const formData = new FormData();
+            formData.append('cover_image', formik.values.cover_image);
+            formData.append('header_image', formik.values.header_image);
+            formData.append('title', formik.values.title);
+            formData.append('blog', formik.values.blog);
+            formData.append('description', formik.values.description);
+            // formData.append('authorization', user.token);
+            fetch(`${paths.BLOG_TOPIC}`, {
+                method: 'POST',
+
+                body: formData,
+            })
+                .then(response => response.json())
+                .then(resp => {
+                    console.log('Success:', resp);
+                })
+                .catch((err) => {
+                    console.error('Error:', err);
+                });
+        },
+        validate: (values) => {
+            const errors = {};
+
+            if (!values.title) {
+                errors.title = 'Required';
+            } else if (values.title.length > 100) {
+                errors.title = 'Too long';
+            }
+            if (!values.blog) {
+                errors.blog = 'Required';
+            }
+
+            return errors;
+        }
+    });
+
+
+    return (
+        <CenteredContainer>
+            <h2>Dodaj kategorię</h2>
+            <StyledForm onSubmit={formik.handleSubmit}>
+                <Input id="title" placeholder="Tytuł" type="text" value={formik.values.title} onChange={event => {
+                    formik.setFieldValue('title', event.target.value);
+                }} />
```

Comment:
te Inputy tutaj generalnie można podmienić na mój komponent jak zmergujesz mastera, a jak nie chcesz to taka ciekawostka że te onChange możesz zamienić na po prostu onChange={formik.handleChange} i będzie działać i tak wszędzie gdzie jest event.target.value ;)